### ISSUE TITLE
Can the C++ MQTT library be built for arm 64?

### ISSUE THREAD

**Issue by sdhir93** [eclipse/paho.mqtt.cpp#316]
I wish to use the MQTT library for a device based on arm 64 architecture.
I compiled the C library with the required compiler - builds fine. But while building the C++ library gives compilation error

/mqtt/paho.mqtt.cpp/src/mqtt/create_options.h:90:9: error: ‘struct MQTTAsync_createOptions’ has no member named ‘allowDisconnectedSendAtAnyTime’
   90 |   opts_.allowDisconnectedSendAtAnyTime = to_int(anyTime);

Basically it is not able to recognize the MQTTAsync.h file. Any suggestions for compiling for Arm 64 ?

**Comment by fpagliughi** [comment]
The latest C++ library requires Paho C v1.3.8

**Comment by sdhir93** [comment]
Yes. I followed the instructions to git checkout the version 1.3.8 for C library and built it in build/_install directory. Pointed to this directory while building the C++ library. Still getting this error.

**Comment by fpagliughi** [comment]
It must be picking up an old version from somewhere. Maybe try asking CMake where it thinks the C headers are living. Try using `-L` to look at the cache:
```
$ cmake -L ..
-- Configuring done
-- Generating done
-- Build files have been written to: /home/fmp/mqtt/paho-mqttpp/build
-- Cache values
CMAKE_BUILD_TYPE:STRING=
CMAKE_INSTALL_PREFIX:PATH=/usr/local
PAHO_BUILD_DEB_PACKAGE:BOOL=OFF
PAHO_BUILD_DOCUMENTATION:BOOL=ON
PAHO_BUILD_SAMPLES:BOOL=ON
PAHO_BUILD_SHARED:BOOL=ON
PAHO_BUILD_STATIC:BOOL=ON
PAHO_BUILD_TESTS:BOOL=OFF
PAHO_MQTT_C_INCLUDE_DIRS:PATH=/home/fmp/mqtt/paho.mqtt.c/build/_install/usr/local/include                <-- ***** --
PAHO_MQTT_C_LIBRARIES:FILEPATH=/home/fmp/mqtt/paho.mqtt.c/build/_install/usr/local/lib/libpaho-mqtt3as.so
PAHO_WITH_SSL:BOOL=ON
```

Is the `PAHO_MQTT_C_INCLUDE_DIRS` entry what you think it should be?

**Comment by sdhir93** [comment]
No, it is pointing to
PAHO_MQTT_C_LIBRARIES:FILEPATH=/usr/local/lib/libpaho-mqtt3as.so

**Comment by sdhir93** [comment]
No, it is pointing to something that I do not want

**Comment by fpagliughi** [comment]
It looks like you followed the instructions for building the library to a non-standard location...
```
To install the library to a non-standard location...
```
Normally, don't use `CMAKE_PREFIX_PATH` and just install to the default location, which on *nix machines would be `/usr/local`.
